appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/ExecutorServicesImpl.java

```diff
@@ -89,7 +98,7 @@ public void cleanup() {
 
     @Override
     protected int getThreadPoolSize() {
-        return executor.getExecutorThreadPoolSize();
+        return Integer.parseInt(this.config.getMaximumPoolSize());
```

Comment:
Javadoc says it is allowed to return `-1` for don't know / unlimited (which is the default anyway).

I'm not sure what this actually optimizes (didn't check Weld), but I feel it is the safer form, and we may skip accessing domain config.

Reply:
As per validation constraints, its min value is `0` and the default value is `Integer.MAX_VALUE`
https://github.com/payara/Payara/blob/master/appserver/concurrent/concurrent-connector/src/main/java/org/glassfish/concurrent/config/ManagedExecutorService.java#L86

Thread pool size is used by Weld to create min tasks in the task factory :
https://github.com/weld/core/blob/master/impl/src/main/java/org/jboss/weld/executor/IterativeWorkerTaskFactory.java#L48

Reply:
So it is only used to set bounds lower than available processors, which is irrelevant in default cause. 

I'm worried how does this behave when one creates an additional non-default managed executor. Which one is picked by the config lookup (or would it fail)?